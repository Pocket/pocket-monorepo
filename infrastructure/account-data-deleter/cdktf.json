{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "90437d3d-e067-4603-8eda-11e297370139"
}