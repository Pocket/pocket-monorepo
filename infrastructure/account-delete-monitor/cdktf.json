{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "d9efd22b-fec2-4561-82e1-9319ba965703"
}