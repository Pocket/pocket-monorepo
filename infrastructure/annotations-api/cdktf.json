{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "5d96b3e5-a2f4-4187-a9a3-2356401e3092"
}