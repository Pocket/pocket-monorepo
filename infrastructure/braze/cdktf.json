{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "d67fe567-bd64-4c5c-b71b-708889a29bb2"
}