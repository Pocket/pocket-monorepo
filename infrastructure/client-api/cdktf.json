{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/main.js",
  "projectId": "7e0b548c-601b-4fd5-9387-cd9c77f5007c"
}
