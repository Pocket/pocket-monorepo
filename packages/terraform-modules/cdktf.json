{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/example.js",
  "projectId": "67d4fb1a-0682-48af-bc23-a5ddc7941efc"
}