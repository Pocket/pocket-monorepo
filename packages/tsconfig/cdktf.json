{
    "$schema": "https://json.schemastore.org/tsconfig",
    "compilerOptions": {
      "experimentalDecorators": true,
      "emitDecoratorMetadata": true,
      "rewriteRelativeImportExtensions": true,
      "allowImportingTsExtensions": true,
      "target": "ES2024",
      "module": "NodeNext",
      "moduleResolution": "NodeNext",
      "removeComments": true,
      "esModuleInterop": true,
      "noEmitOnError": false,
      "sourceMap": true,
      "inlineSources": true,
      "sourceRoot": "/",
      "declaration": true,
      "strictNullChecks": true
    }
  }