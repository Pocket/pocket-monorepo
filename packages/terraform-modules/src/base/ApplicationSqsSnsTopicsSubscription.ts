import {
  dataAwsIamPolicyDocument,
  snsTopicSubscription,
  sqsQueue,
  sqsQueuePolicy,
  dataAwsSqsQueue,
} from '@cdktf/provider-aws';
import { type SnsTopicSubscriptionConfig } from '@cdktf/provider-aws/lib/sns-topic-subscription';
import { TerraformMetaArguments, TerraformResource } from 'cdktf';
import { Construct } from 'constructs';

export interface SnsSqsSubscriptionProps {
  name: string;
  snsTopicArn: string;
  snsDlq?: sqsQueue.SqsQueue;
  filterPolicy?: SnsTopicSubscriptionConfig['filterPolicy'];
  filterPolicyScope?: SnsTopicSubscriptionConfig['filterPolicyScope'];
}

export interface ApplicationSqsSnsTopicsSubscriptionProps
  extends TerraformMetaArguments {
  subscriptions: SnsSqsSubscriptionProps[];
  name: string;
  sqsQueue: sqsQueue.SqsQueue | dataAwsSqsQueue.DataAwsSqsQueue;
  tags?: { [key: string]: string };
  dependsOn?: TerraformResource[];
}

/**
 * Creates an SNS to SQS subscription, allowing an SQS queue to
 * subscribe to multiple topics (in the rare case where this pattern
 * is useful)
 */
export class ApplicationSqsSnsTopicsSubscription extends Construct {
  public readonly snsTopicSubscriptions: snsTopicSubscription.SnsTopicSubscription[];

  constructor(
    scope: Construct,
    name: string,
    private config: ApplicationSqsSnsTopicsSubscriptionProps,
  ) {
    super(scope, name);
    const subscriptions = config.subscriptions.map((sub) => ({
      ...sub,
      snsDlq: sub.snsDlq ?? this.createSqsSubscriptionDlq(sub.name),
    }));
    this.snsTopicSubscriptions = subscriptions.map((sub) => {
      return this.createSnsTopicSubscription(sub);
    });
    this.createPoliciesForSnsToSQS(subscriptions);
  }

  /**
   * Create a dead-letter queue for failed SNS messages
   * @private
   */
  private createSqsSubscriptionDlq(name: string): sqsQueue.SqsQueue {
    return new sqsQueue.SqsQueue(this, `${name}-sns-topic-dql`, {
      name: `${name}-SNS-Topic-DLQ`,
      tags: this.config.tags,
      provider: this.config.provider,
    });
  }

  /**
   * Create an SNS subscription for SQS
   * @param snsTopicDlq
   * @private
   */
  private createSnsTopicSubscription(
    properties: Omit<SnsSqsSubscriptionProps, 'snsDlq'> & {
      snsDlq: sqsQueue.SqsQueue;
    },
  ): snsTopicSubscription.SnsTopicSubscription {
    return new snsTopicSubscription.SnsTopicSubscription(
      this,
      `${properties.name}-sns-subscription`,
      {
        topicArn: properties.snsTopicArn,
        protocol: 'sqs',
        endpoint: this.config.sqsQueue.arn,
        redrivePolicy: JSON.stringify({
          deadLetterTargetArn: properties.snsDlq.arn,
        }),
        filterPolicy: properties.filterPolicy,
        filterPolicyScope: properties.filterPolicyScope,
        dependsOn: [
          properties.snsDlq,
          ...(this.config.dependsOn ? this.config.dependsOn : []),
        ],
        provider: this.config.provider,
      } as snsTopicSubscription.SnsTopicSubscriptionConfig,
    );
  }

  /**
   * Create IAM policies to allow SNS to write to the target SQS queue and a
   * dead-letter queue
   * @param snsTopicDlq
   * @private
   */
  private createPoliciesForSnsToSQS(
    subscriptions: Array<
      Omit<SnsSqsSubscriptionProps, 'snsDlq'> & {
        snsDlq: sqsQueue.SqsQueue;
      }
    >,
  ): void {
    // Make DLQ policies first since they are separate
    subscriptions.forEach((sub) => {
      const policy = new dataAwsIamPolicyDocument.DataAwsIamPolicyDocument(
        this,
        `${sub.name}-sns-dlq-policy-document`,
        {
          statement: [
            {
              effect: 'Allow',
              actions: ['sqs:SendMessage'],
              resources: [sub.snsDlq.arn],
              principals: [
                {
                  identifiers: ['sns.amazonaws.com'],
                  type: 'Service',
                },
              ],
              condition: [
                {
                  test: 'ArnEquals',
                  variable: 'aws:SourceArn',
                  values: [sub.snsTopicArn],
                },
              ],
            },
          ],
          dependsOn: [sub.snsDlq] as TerraformResource[],
          provider: this.config.provider,
        },
      ).json;

      return new sqsQueuePolicy.SqsQueuePolicy(
        this,
        `${sub.name}-sns-dlq-policy`,
        {
          queueUrl: sub.snsDlq.url,
          policy: policy,
          provider: this.config.provider,
        },
      );
    });

    const queuePolicyDoc =
      new dataAwsIamPolicyDocument.DataAwsIamPolicyDocument(
        this,
        `${this.config.name}-sns-sqs-policy-document`,
        {
          statement: [
            {
              effect: 'Allow',
              actions: ['sqs:SendMessage'],
              resources: [this.config.sqsQueue.arn],
              principals: [
                {
                  identifiers: ['sns.amazonaws.com'],
                  type: 'Service',
                },
              ],
              condition: [
                {
                  test: 'ArnEquals',
                  variable: 'aws:SourceArn',
                  values: subscriptions.map((sub) => sub.snsTopicArn),
                },
              ],
            },
          ],
          dependsOn: [this.config.sqsQueue] as TerraformResource[],
          provider: this.config.provider,
        },
      ).json;

    new sqsQueuePolicy.SqsQueuePolicy(
      this,
      `${this.config.name}-sns-sqs-policy`,
      {
        queueUrl: this.config.sqsQueue.url,
        policy: queuePolicyDoc,
        provider: this.config.provider,
      },
    );
  }
}
