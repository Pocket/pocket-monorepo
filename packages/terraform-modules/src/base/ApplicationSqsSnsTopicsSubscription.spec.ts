import { sqsQueue } from '@cdktf/provider-aws';
import { Testing } from 'cdktf';
import { ApplicationSqsSnsTopicsSubscription } from './ApplicationSqsSnsTopicsSubscription.ts';

describe('ApplicationSqsSnsTopicSubscription', () => {
  const getConfig = (stack) => ({
    name: 'test-sns-subscription',
    subscriptions: [
      { name: 'TopicName', snsTopicArn: 'arn:aws:sns:TopicName' },
      { name: 'TopicSub2', snsTopicArn: 'arn:aws:sns:AnotherTopic' },
    ],
    sqsQueue: new sqsQueue.SqsQueue(stack, 'sqs', {
      name: 'test-sqs',
    }),
  });

  const getConfigWithDlq = (stack) => ({
    name: 'test-sns-subscription',
    subscriptions: [
      {
        snsTopicArn: 'arn:aws:sns:TopicName',
        name: 'TopicSub',
        snsDlq: new sqsQueue.SqsQueue(stack, 'dlq', {
          name: 'test-sqs-dlq',
        }),
      },
    ],
    sqsQueue: new sqsQueue.SqsQueue(stack, 'sqs', {
      name: 'test-sqs',
    }),
  });

  it('renders an SQS SNS subscription without tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSqsSnsTopicsSubscription(
        stack,
        'sqs-sns-subscription',
        getConfig(stack),
      );
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an SQS SNS subscription with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSqsSnsTopicsSubscription(stack, 'sqs-sns-subscription', {
        ...getConfig(stack),
        tags: { hello: 'there' },
      });
    });
    expect(synthed).toMatchSnapshot();
  });

  it('renders an SQS SNS subscription with dlq passed', () => {
    const synthed = Testing.synthScope((stack) => {
      new ApplicationSqsSnsTopicsSubscription(stack, 'sqs-sns-subscription', {
        ...getConfigWithDlq(stack),
        tags: { hello: 'there' },
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});
