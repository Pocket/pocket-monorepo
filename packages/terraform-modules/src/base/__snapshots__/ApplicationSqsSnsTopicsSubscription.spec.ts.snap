// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationSqsSnsTopicSubscription renders an SQS SNS subscription with dlq passed 1`] = `
"{
  "data": {
    "aws_iam_policy_document": {
      "sqs-sns-subscription_TopicSub-sns-dlq-policy-document_07666DAB": {
        "depends_on": [
          "aws_sqs_queue.dlq"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.dlq.arn}"
            ]
          }
        ]
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A": {
        "depends_on": [
          "aws_sqs_queue.sqs"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs.arn}"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_sns_topic_subscription": {
      "sqs-sns-subscription_TopicSub-sns-subscription_46BF1FB9": {
        "depends_on": [
          "aws_sqs_queue.dlq"
        ],
        "endpoint": "\${aws_sqs_queue.sqs.arn}",
        "protocol": "sqs",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.dlq.arn}\\"}",
        "topic_arn": "arn:aws:sns:TopicName"
      }
    },
    "aws_sqs_queue": {
      "dlq": {
        "name": "test-sqs-dlq"
      },
      "sqs": {
        "name": "test-sqs"
      }
    },
    "aws_sqs_queue_policy": {
      "sqs-sns-subscription_TopicSub-sns-dlq-policy_5A6F202C": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_TopicSub-sns-dlq-policy-document_07666DAB.json}",
        "queue_url": "\${aws_sqs_queue.dlq.url}"
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy_152AA94D": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A.json}",
        "queue_url": "\${aws_sqs_queue.sqs.url}"
      }
    }
  }
}"
`;

exports[`ApplicationSqsSnsTopicSubscription renders an SQS SNS subscription with tags 1`] = `
"{
  "data": {
    "aws_iam_policy_document": {
      "sqs-sns-subscription_TopicName-sns-dlq-policy-document_921B87EF": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.arn}"
            ]
          }
        ]
      },
      "sqs-sns-subscription_TopicSub2-sns-dlq-policy-document_FD0E0FAB": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:AnotherTopic"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.arn}"
            ]
          }
        ]
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A": {
        "depends_on": [
          "aws_sqs_queue.sqs"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName",
                  "arn:aws:sns:AnotherTopic"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs.arn}"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_sns_topic_subscription": {
      "sqs-sns-subscription_TopicName-sns-subscription_1AD59E90": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC"
        ],
        "endpoint": "\${aws_sqs_queue.sqs.arn}",
        "protocol": "sqs",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.arn}\\"}",
        "topic_arn": "arn:aws:sns:TopicName"
      },
      "sqs-sns-subscription_TopicSub2-sns-subscription_68D4A411": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6"
        ],
        "endpoint": "\${aws_sqs_queue.sqs.arn}",
        "protocol": "sqs",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.arn}\\"}",
        "topic_arn": "arn:aws:sns:AnotherTopic"
      }
    },
    "aws_sqs_queue": {
      "sqs": {
        "name": "test-sqs"
      },
      "sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC": {
        "name": "TopicName-SNS-Topic-DLQ",
        "tags": {
          "hello": "there"
        }
      },
      "sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6": {
        "name": "TopicSub2-SNS-Topic-DLQ",
        "tags": {
          "hello": "there"
        }
      }
    },
    "aws_sqs_queue_policy": {
      "sqs-sns-subscription_TopicName-sns-dlq-policy_F31EB07E": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_TopicName-sns-dlq-policy-document_921B87EF.json}",
        "queue_url": "\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.url}"
      },
      "sqs-sns-subscription_TopicSub2-sns-dlq-policy_77228AB6": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_TopicSub2-sns-dlq-policy-document_FD0E0FAB.json}",
        "queue_url": "\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.url}"
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy_152AA94D": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A.json}",
        "queue_url": "\${aws_sqs_queue.sqs.url}"
      }
    }
  }
}"
`;

exports[`ApplicationSqsSnsTopicSubscription renders an SQS SNS subscription without tags 1`] = `
"{
  "data": {
    "aws_iam_policy_document": {
      "sqs-sns-subscription_TopicName-sns-dlq-policy-document_921B87EF": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.arn}"
            ]
          }
        ]
      },
      "sqs-sns-subscription_TopicSub2-sns-dlq-policy-document_FD0E0FAB": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:AnotherTopic"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.arn}"
            ]
          }
        ]
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A": {
        "depends_on": [
          "aws_sqs_queue.sqs"
        ],
        "statement": [
          {
            "actions": [
              "sqs:SendMessage"
            ],
            "condition": [
              {
                "test": "ArnEquals",
                "values": [
                  "arn:aws:sns:TopicName",
                  "arn:aws:sns:AnotherTopic"
                ],
                "variable": "aws:SourceArn"
              }
            ],
            "effect": "Allow",
            "principals": [
              {
                "identifiers": [
                  "sns.amazonaws.com"
                ],
                "type": "Service"
              }
            ],
            "resources": [
              "\${aws_sqs_queue.sqs.arn}"
            ]
          }
        ]
      }
    }
  },
  "resource": {
    "aws_sns_topic_subscription": {
      "sqs-sns-subscription_TopicName-sns-subscription_1AD59E90": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC"
        ],
        "endpoint": "\${aws_sqs_queue.sqs.arn}",
        "protocol": "sqs",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.arn}\\"}",
        "topic_arn": "arn:aws:sns:TopicName"
      },
      "sqs-sns-subscription_TopicSub2-sns-subscription_68D4A411": {
        "depends_on": [
          "aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6"
        ],
        "endpoint": "\${aws_sqs_queue.sqs.arn}",
        "protocol": "sqs",
        "redrive_policy": "{\\"deadLetterTargetArn\\":\\"\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.arn}\\"}",
        "topic_arn": "arn:aws:sns:AnotherTopic"
      }
    },
    "aws_sqs_queue": {
      "sqs": {
        "name": "test-sqs"
      },
      "sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC": {
        "name": "TopicName-SNS-Topic-DLQ"
      },
      "sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6": {
        "name": "TopicSub2-SNS-Topic-DLQ"
      }
    },
    "aws_sqs_queue_policy": {
      "sqs-sns-subscription_TopicName-sns-dlq-policy_F31EB07E": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_TopicName-sns-dlq-policy-document_921B87EF.json}",
        "queue_url": "\${aws_sqs_queue.sqs-sns-subscription_TopicName-sns-topic-dql_785B69DC.url}"
      },
      "sqs-sns-subscription_TopicSub2-sns-dlq-policy_77228AB6": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_TopicSub2-sns-dlq-policy-document_FD0E0FAB.json}",
        "queue_url": "\${aws_sqs_queue.sqs-sns-subscription_TopicSub2-sns-topic-dql_6F6C03F6.url}"
      },
      "sqs-sns-subscription_test-sns-subscription-sns-sqs-policy_152AA94D": {
        "policy": "\${data.aws_iam_policy_document.sqs-sns-subscription_test-sns-subscription-sns-sqs-policy-document_9FB65D3A.json}",
        "queue_url": "\${aws_sqs_queue.sqs.url}"
      }
    }
  }
}"
`;
